repository.rst

```diff
@@ -156,6 +156,12 @@ returned by ``go env GOPATH``.
 | A replacement for the module named by ``importpath``. The module named by                                                 |
 | ``replace`` will be downloaded at ``version`` and verified with ``sum``.                                                  |
 |                                                                                                                           |
+| NOTE: Replace does not work with `commit` or `tag`, or other git selection params                                         |
+| since in this mode, bazel doesn't need to resolve dependencies like go does                                               |
+| where replace would make a difference. To get replace behaviour simply change                                             |
+| the importpath in a rule downloading the dependency without a replace                                                     |
+| replace attribute. See the "Download from git fork" example above.                                                        |
```

Comment:
Rephrase and format:

```
NOTE: ``replace`` only works in module mode and cannot be used with ``commit``,
``tag``, or other VCS attributes. To download directly from a Git repository,
see the "Download from git fork" example above.
```

It is possible to download from a forked repo in module mode though. So it might be better to add another example as well (and mention it here):

```
# Download from a module fork
go_repository(
    name = "com_github_pkg_errors",
    importpath = "github.com/pkg/errors",
    replace = "github.com/fork/errors",
    sum = "h1:iURUrRGxPUNPdy5/HRSm+Yj6okJ6UtLINN0Q9M4+h3I=",
    version = "v0.8.1-forkedversion",
)
```

Reply:
First off sorry for the delay in this, I don't see it in my Github Notifcation box at all :/ not sure what happened there.

I don't like this particular rephrasing though, the particular brunt of my frustration that brought around this PR is the lack of really thinking about _why_ replace doesn't actually even need to work, and why it doesn't work. This rephrasing simply says: "This doesn't work" which isn't actually helpful if you're not familiar with why it doesn't work (and why it really in effect doesn't need too):

```
NOTE: `replace` only affects things in module mode and cannot be used with ``commit``,
``tag``, or other VCS attributes. If using VCS attributes you should simply specify a seperate
importpath.
```

Reply:
I'd change the last sentence to 

```
When using VCS attributes, you can use the ``remote`` attribute instead.
See the "Download from git fork" example above.
```

`importpath` is the main attribute of `go_repository`. It is a prefix for the `importpath` of every generated `go_library` target in the repo. Changing it means you're fetching a different module.

`replace` and `remote` are used to change where the module is downloaded from (in module mode and VCS mode, respectively). Normally, `importpath` determines that, but `replace` and `remote` let you download the same module (or a modified version with the same import paths) from a different location.

Reply:
Should be fixed in the latest patch!

Reply:
Wording looks good. Please format code with two quotes:

```
``like this`` not `like this`
```